### PR TITLE
feat: Add BlockUserUseCase

### PR DESCRIPTION
**Pull request by borichellow**
# Needed for [Task-1593](https://wearezeta.atlassian.net/browse/AR-1593)

# What's new in this PR?

added BlockUserUseCase and tests for it.

### CODE REVIEW

**Review by MohamadJaara — APPROVED** [review]
lgtm 🚀

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/connection/BlockUserUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+internal class BlockUserUseCaseImpl(
+    private val connectionRepository: ConnectionRepository
+) : BlockUserUseCase {
+
+    override suspend fun invoke(userId: UserId): BlockUserResult {
+        return connectionRepository.updateConnectionStatus(userId, ConnectionState.BLOCKED)
+            .fold({
+                kaliumLogger.e("An error occurred when blocking a user $userId")
```
Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/connection/BlockUserUseCase.kt:30]: Let's also log the error?

**Review by typfel — COMMENTED** [review]
Looks good to me, just a comment on the logging.

**Review by typfel — APPROVED** [review]

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/connection/BlockUserUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.connection.ConnectionRepository
+import com.wire.kalium.logic.data.user.ConnectionState
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.logic.kaliumLogger
+
+/**
+ * Use Case that allows a user to ignore a connection request from given user
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/connection/BlockUserUseCase.kt:11]:
```suggestion
 * Use Case that allows a user to block a connection request from given user
```

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/connection/BlockUserUseCaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import io.mockative.classOf
+import io.mockative.given
+import io.mockative.mock
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertTrue
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class BlockUserUseCaseTest {
```
Comment (yamilmedina) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/connection/BlockUserUseCaseTest.kt:21]: ❤️

**Review by yamilmedina — APPROVED** [review]
🥇

**Pull request merged**